elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -855,7 +860,7 @@ public PersistentResource getRelation(String relation, String id) {
         Set<String> missedIds = Sets.difference(new HashSet<>(ids), allExpectedIds);
 
         if (!missedIds.isEmpty()) {
-            throw new InvalidObjectIdentifierException(missedIds.toString(), relation);
+            throw new EntityForbiddenException();
```

Comment:
If we are not going to return the missing IDs, we can remove the calculation of allExpectedIds and missedIds above.

Reply:
The `missedIds` is still used to check if queried id matches ant existent entity id.
```
if (!missedIds.isEmpty()) {
...

```

Reply:
You can remove if you change the check to:
```
if (ids.size() != allResources.size()) {
```

Reply:
```
if (ids.size() != allResources.size()) {
```
I think this only checks their cardinality. Even the sizes are the same, their id could be mismatched. Say a query `/book/1000` to a entity `author` who has one book with id 1 will not trigger the above statement.